src/main/java/com/peerrequest/app/api/EntriesController.java

```diff
@@ -0,0 +1,98 @@
+
+/**
+ * The controller class for entries to encapsulate the functionalities of the corresponding
+ * service and provide them as HTTP-endpoints.
+ */
+@RestController
+@ApiControllerPrefix
+public class EntriesController extends ServiceBasedController {
+
+    private final int maxPageSize = 100;
+
+    @GetMapping("/categories/{category_id}/entries")
+    List<Entry.Dto> listEntries(@RequestParam Optional<Integer> limit,
+                                @RequestParam Optional<Long> after,
+                                @PathVariable Long category_id) {
+        if (limit.isPresent()) {
+            if (limit.get() < 0) {
+                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "limit must be greater than 0");
+            }
+            limit = Optional.of(Math.min(limit.get(), maxPageSize));
+        }
+
+        Entry.Dto filterEntry = new Entry.Dto(
+                null, null, null, null, Optional.of(category_id));
+
+        return this.entryService.list(after.orElse(null), limit.orElse(maxPageSize), filterEntry)
+                .stream().map(Entry::toDto).toList();
+    }
+
+    @GetMapping("/categories/{category_id}/entries/{entry_id}")
+    Entry.Dto getEntry(@PathVariable(name = "entry_id") Long id) {
+        var option = this.entryService.get(id);
+        if (option.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "entry does not exist");
+        }
+
+        return option.get().toDto();
+    }
+
+    @GetMapping("/categories/{category_id}/entries/{entry_id}/paper")
+    void getPaper(@PathVariable(name = "entry_id") Long id) {
+        // TODO: Implement after DocumentService is finished
+        throw new RuntimeException("Not implemented yet");
+    }
+
+    @PostMapping("/categories/{category_id}/entries")
+    Entry.Dto createEntries(@RequestBody Entry.Dto dto,
+                            @AuthenticationPrincipal OAuth2User user, @PathVariable Long category_id) {
+        if (dto.name() == null) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "name is required");
+        }
+
+        // TODO: Only allow entry creation if category exists AND user is authorized
+
+        if (dto.documentId() == null) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "document_id is required");
+        }
+
+        var entry = Entry.fromDto(dto, user.getAttribute("sub"), category_id);
+        return this.entryService.create(entry.toDto()).toDto();
+    }
+
+    @PatchMapping("/categories/{category_id}/entries")
+    Entry.Dto patchEntries(@RequestBody Entry.Dto dto, @AuthenticationPrincipal OAuth2User user) {
+        if (dto.id().isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "id is required");
+        }
+
+        if (dto.researcherId().isPresent()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "you may not change the researcher_id");
+        }
+
```

Comment:
check is missing if document id is present

Reply:
Added the check.